networks/rpc/types_test.go

```diff
@@ -42,7 +43,7 @@ func TestBlockNumberJSONUnmarshal(t *testing.T) {
 		6:  {`"0x12"`, false, BlockNumber(18)},
 		7:  {`"0x7fffffffffffffff"`, false, BlockNumber(math.MaxInt64)},
 		8:  {`"0x8000000000000000"`, true, BlockNumber(0)},
-		9:  {"0", true, BlockNumber(0)},
+		9:  {"0x0", true, BlockNumber(0)},
```

Comment:
Why did you change this value?

Reply:
`9:  {"0", true, BlockNumber(0)},`
This code is fail test when a decimal input comes.
I've changed it because I think it's unnecessary.

Now thought, this can  be as a successful unit test.
I added some unit tests including this test .
Thanks!